src/framedata.cpp

```diff
@@ -0,0 +1,52 @@
+#include "framedata.h"
+
+#include "completion.h"
+#include "layout.h"
+
+using std::string;
+
+size_t layoutAlgorithmCount() {
+    size_t i = 0;
+    while (g_layout_names[i] != nullptr) {
+        i++;
+    }
+    return i;
+}
+
+template<> LayoutAlgorithm Converter<LayoutAlgorithm>::parse(const string& source, LayoutAlgorithm const* relativeTo) {
+    for (size_t i = 0; g_layout_names[i] != nullptr; i++) {
+        if (source == g_layout_names[i]) {
+            return (LayoutAlgorithm) i;
+        }
+    }
+    throw std::invalid_argument("Invalid split align name: \"" + source + "\"");
+}
+
+template<> string Converter<LayoutAlgorithm>::str(LayoutAlgorithm payload) {
+    return string(g_layout_names[(int)payload]);
```

Comment:
The `string(...)` should not be necessary here (also, a space after the cast would be nice).

The same applies to the `str` method

Reply:
I've fixed this (and updated this to the new Converter interface, cf PR #436)